common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java

```diff
@@ -43,7 +45,13 @@ public String[] regexMatch(@LuaNotNil String str, @LuaNotNil String pattern) {
             matches.add(matcher.group());
         }
 
-        return matches.toArray(new String[0]);
+        LuaTable tbl = new LuaTable();
```

Comment:
preallocate table pls
```suggestion
        LuaTable tbl = new LuaTable(matches.size());
```